ugs-core/src/com/willwinder/universalgcodesender/listeners/ControllerListenerAdapter.java

```diff
@@ -0,0 +1,38 @@
+package com.willwinder.universalgcodesender.listeners;
+
+import com.willwinder.universalgcodesender.model.Position;
+import com.willwinder.universalgcodesender.model.UGSEvent;
+import com.willwinder.universalgcodesender.types.GcodeCommand;
+
+public class ControllerListenerAdapter implements ControllerListener {
```

Comment:
I see what you did now. `ControllerListener` is sort of deprecated because I don't like having all the unused methods either. You can use `ControllerStateListener` to get this data instead and it utilizes the same `UGSEvent` callback.

Reply:
An Adapter is common pattern to use when only one or a few of the interface methods are interesting to implement, then you don't have all those blank methods. This was purely to reduce the number of lines of code. I'll look into ControllerStateListener. I was afraid of breaking functionality here.

Reply:
Marking as done for tracking my tasks.